### PR TITLE
Add an e2e test to validate fault injection telemetry.

### PR DESCRIPTION
**Pull request by douglas-reid**
This attempts to provide validation of telemetry for FI to guard against
recurrence of issues such as: https://github.com/istio/istio/issues/11151.

It adds a new test in the mixer suite that installs custom virtual
service and destination rules that inject faults at 100% (using error code 555).

The test validates that the destination workload information is
"unknown" and that we receive telemetry with the `FI` response flag.

Ran this on a test cluster:

```
--- PASS: TestFaultInjectionTelemetry (59.17s)
    mixer_test.go:918: Sending traffic...
    retry.go:86: Trying to find metrics via promql (attempt 1)...
    retry.go:86: prometheus query: sum(istio_requests_total{response_code="555", response_flags="FI"}) by (destination_workload, destination_app)
PASS

```

### CODE REVIEW

File: tests/e2e/tests/mixer/mixer_test.go [hunk 1]
```diff
@@ -938,6 +939,75 @@ func sendTraffic(t *testing.T, msg string, calls int64) *fhttp.HTTPRunnerResults
...
+	if got, want := len(vector), 1; got != want {
+		t.Errorf("got %d different labels for destination_workload and destination_app, want %d", got, want)
+	}
+
+	got, err := vectorValue(vector, map[string]string{"destination_workload": "unknown", "destination_app": "unknown"})
+	if err != nil {
+		t.Errorf("could not get extract value from: %#v: %v", vector, err)
+	}
+
+	want := 50.0
```
Comment (utka) [inline tests/e2e/tests/mixer/mixer_test.go]: Why want is 50 and not 100 if  sendTraffic(..., 100)?

Comment (douglas-reid) [reply tests/e2e/tests/mixer/mixer_test.go]: Two reasons:
1. This test isn't trying to validate precise networking behavior here (including retries, etc. that may be configured)
1. Prior experience with test flakiness when trying to be too precise for portions of the test that aren't critical.

This is a (somewhat arbitrary) minimum bar to ensure that we have a proper sample size for validating that the previously buggy behavior doesn't exist.  If we only had one request, that isn't really good enough, unfortunately.

Does that make sense?

**Comment by douglas-reid** [comment]
Kindly ping, reviewers.

**Comment by douglas-reid** [comment]
@rshriram would you mind kindly granting samples/OWNERS approval?

File: samples/bookinfo/networking/fault-injection-details-v1.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: details
+spec:
+  host: details
+  subsets:
+  - name: v1
+    labels:
+      version: v1
```
Comment (rshriram) [inline samples/bookinfo/networking/fault-injection-details-v1.yaml]: not needed..

File: samples/bookinfo/networking/fault-injection-details-v1.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        httpStatus: 555
+        percent: 100
+    route:
+    - destination:
+        host: details
+        subset: v1
+  - route:
+    - destination:
+        host: details
+        subset: v1
```
Comment (rshriram) [inline samples/bookinfo/networking/fault-injection-details-v1.yaml]: no need for subset

Comment (rshriram) [reply samples/bookinfo/networking/fault-injection-details-v1.yaml]: no need for the second route

**Pull request merged**